src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -179,10 +170,15 @@ object RuleUtils {
     // If there is no change in source data files, the index can be applied by
     // transformPlanToUseIndexOnlyScan regardless of Hybrid Scan config.
     // This tag should always exist if Hybrid Scan is enabled.
-    lazy val hybridScanRequired =
+    val hybridScanRequired = HyperspaceConf.hybridScanEnabled(spark) &&
       index.getTagValue(getLogicalRelation(plan).get, IndexLogEntryTags.HYBRIDSCAN_REQUIRED).get
 
-    val transformed = if (HyperspaceConf.hybridScanEnabled(spark) && hybridScanRequired) {
+    // In case the index has appended files and/or deleted files which are updated by quick
+    // refresh mode, we should handle them in query time by using Hybrid Scan.
+    // Because quick refresh doesn't update index data, but just metadata in IndexLogEntry.
+    lazy val quickRefreshedIndex = index.hasSourceUpdate
```

Comment:
If we are using it only once, do we need to create a separate (lazy) val for it?

Reply:
It's mainly for readability and comments. + It's only required when HybridScan is enabled, so it's lazy val.

Reply:
I would add a comment if you added `lazy` to avoid creating sets for appended files, etc.